bundles/storage/org.eclipse.smarthome.storage.json/src/main/java/org/eclipse/smarthome/storage/json/ConfigurationDeserializer.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.storage.json;
+
+import java.lang.reflect.Type;
+import java.math.BigDecimal;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map.Entry;
+
+import org.eclipse.smarthome.config.core.Configuration;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonPrimitive;
+
+/**
+ * Deserializes a {@link Configuration} object.
+ *
+ * As opposed to Gson's default behavior, it ensures that all numbers are represented as {@link BigDecimal}s.
+ *
+ * @author Simon Kaufmann - initial contribution and API
+ *
+ */
+public class ConfigurationDeserializer implements JsonDeserializer<Configuration> {
+
+    @Override
+    public Configuration deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
+            throws JsonParseException {
+        Configuration configuration = new Configuration();
+        JsonObject configurationObject = json.getAsJsonObject();
+        JsonObject propertiesObject = configurationObject.get("properties").getAsJsonObject();
```

Comment:
Shouldn't we be "save" and test if `configurationObject.get("properties")` returns `null`?